### PR TITLE
refactor(tests): simplify test_logs.py with fixtures and parameterization

### PR DESCRIPTION
**Pull request by major**
- Add pytest fixtures with proper specs (mock_execute_command, mock_path_exists, mock_allowed_log_paths) replacing inline mocking
- Use pytest.mark.parametrize to consolidate duplicate test cases (e.g., 6 filter tests → 1 parameterized test)
- Add mock specs (spec=execute_command) for type safety verification
- Extract shared assertion logic into assert_tool_result_structure()

### CODE REVIEW

File: tests/tools/test_logs.py [hunk 1]
```diff
@@ -1,641 +1,463 @@
...
-        output = result[0][0].text
+@pytest.fixture
+def mock_execute_command(mocker):
+    """Mock execute_command with proper spec."""
+    mock = mocker.patch(
+        "linux_mcp_server.tools.logs.execute_command",
+        spec=execute_command,
+        new_callable=AsyncMock,
+    )
+    return mock
```
Comment (r0x0d) [inline tests/tools/test_logs.py]: Can probably be moved to conftest, since there is a lot of files that use this

Comment (major) [reply tests/tools/test_logs.py]: Good point. I'll adjust that.

File: tests/tools/test_logs.py [hunk 2]
```diff
@@ -1,641 +1,450 @@
...
-        )
+from linux_mcp_server.server import mcp
 
-        result = await mcp.call_tool("get_journal_logs", {"priority": "err"})
 
-        assert isinstance(result, tuple)
-        assert len(result) == 2
-        assert isinstance(result[0], list)
-        output = result[0][0].text
+async def assert_tool_result_structure(result):
```
Comment (r0x0d) [inline tests/tools/test_logs.py]:
```suggestion
def assert_tool_result_structure(result):
```

no need for it to be async, right?

File: tests/tools/test_logs.py [hunk 3]
```diff
@@ -1,641 +1,450 @@
...
 
-        assert "priority=err" in output
-        assert "Error occurred" in output
 
-        # Verify priority filter was passed
-        args = mock_execute_command.call_args[0][0]
-        assert "-p" in args
-        assert "err" in args
+@pytest.fixture
+def mock_path_exists(mocker):
```
Comment (r0x0d) [inline tests/tools/test_logs.py]: Sorry, I don't think that this fixture provides much value to the test module...

File: tests/tools/test_logs.py (hunk 2 above)
Comment (r0x0d) [reply tests/tools/test_logs.py]: Since this is a common used assertion for all tools, I think in a follow-up PR we could move this to a different module and re-use that

File: tests/tools/test_logs.py [hunk 4]
```diff
@@ -1,641 +1,450 @@
...
-        assert "No journal entries found matching the criteria" in output
-
-    async def test_get_journal_logs_command_error(self, mocker):
-        """Test get_journal_logs handles command errors."""
-        mock_execute_command = mocker.patch("linux_mcp_server.tools.logs.execute_command")
-        mock_execute_command.return_value = (1, "", "journalctl: Failed to access journal")
-
-        result = await mcp.call_tool("get_journal_logs", {})
+        result = await mcp.call_tool("get_journal_logs", params)
+        output = await assert_tool_result_structure(result)
```
Comment (r0x0d) [inline tests/tools/test_logs.py]:
```suggestion
        output =  assert_tool_result_structure(result)
```

we don't need to await on this assert because we have already awaited the result above

File: tests/tools/test_logs.py [hunk 5]
```diff
@@ -1,641 +1,450 @@
...
+    )
+    async def test_get_journal_logs_edge_cases(
+        self, mock_execute_command, returncode, stdout, stderr, expected_error, expectation
+    ):
+        """Test get_journal_logs error handling and edge cases."""
+        mock_execute_command.return_value = (returncode, stdout, stderr)
+
+        with expectation:
+            result = await mcp.call_tool("get_journal_logs", {})
+            output = await assert_tool_result_structure(result)
```
Comment (r0x0d) [inline tests/tools/test_logs.py]: same here

File: tests/conftest.py [hunk 6]
```diff
@@ -211,3 +211,18 @@ async def list_services(*args, **kwargs):
...
+
+@pytest.fixture
+def mock_execute_command(mocker):
+    """Mock execute_command with proper spec."""
+    from unittest.mock import AsyncMock
+
+    from linux_mcp_server.connection.ssh import execute_command
+
+    mock = mocker.patch(
+        "linux_mcp_server.tools.logs.execute_command",
```
Comment (r0x0d) [inline tests/conftest.py:227]: If this will be here available to all others, then we can't mock only the `logs.execute_command`, but `connection.sh.exececute_command`

File: tests/conftest.py [hunk 7]
```diff
@@ -211,3 +211,18 @@ async def list_services(*args, **kwargs):
         raise ValueError("Raised intentionally")
 
     return list_services
+
+
+@pytest.fixture
+def mock_execute_command(mocker):
+    """Mock execute_command with proper spec."""
+    from unittest.mock import AsyncMock
```
Comment (r0x0d) [inline tests/conftest.py]: This can be moved to the top. No reason to have it in the function

File: tests/conftest.py [hunk 8]
```diff
@@ -211,3 +211,18 @@ async def list_services(*args, **kwargs):
...
+
+    from linux_mcp_server.connection.ssh import execute_command
```
Comment (r0x0d) [inline tests/conftest.py]: Add a comment on why we are importing in here

File: tests/tools/test_logs.py (hunk 3 above)
Comment (major) [reply tests/tools/test_logs.py]: Yeah, it's pretty basic. I'll move it back to inline

File: tests/tools/test_logs.py (hunk 4 above)
Comment (major) [reply tests/tools/test_logs.py]: Oh good point. I missed that.
